### DISCUSSION TITLE
Maximale Stromstärke Wallbox

### DISCUSSION THREAD

**Discussion by JoernHei** [camueller/SmartApplianceEnabler#338] [Installation und Betrieb]
Hallo camueller,
vorweg erstmal vielen Dank für die tolle Umsetzung des SmartApplianceEnablers.
Ich habe bereits seit einiger Zeit die Steuerung einer Wärmepumpe und von zwei Zählern damit realisiert.
Jetzt wollte ich noch den go-e Charger mit einfügen. Jedoch habe ich eine Frage zu der dem SHM gemeldeten maximalen Leistung: Diese wird einmal durch die Wallbox und einmal durch das Fahrzeug begrenzt. Der SHM müsste doch mit der Leistung zwischen dem Mindestwert und dem maximal begrenzenden Faktor (Auto oder Wallbox) rechnen. In der xml Datei mit den Daten für den SHM wird jedoch nur der maximale Wert der Wallbox gesetzt. Dieser entspricht nicht immer dem des Fahrzeugs.
Bei mir kann die Wallbox 22 kW liefern, das eine Auto jedoch nur mit 7,2 kW und das andere mit lediglich 3,4 kW laden. Bei einer Planung mit hohem Überschuss befürchte ich, dass das Laden zu spät startet, und das Auto nicht die gewünschte Menge an Energie in der verbleibenden Zeit in den Akku bekommt.
Anbei habe ich das XML von "raspi/semp" angefügt. Zur besseren Übersicht habe ich alle anderen Devices eingeklappt und die Device ID ist mit "ÄÄÄÄÄ" ersetzt. [xml.txt](https://github.com/camueller/SmartApplianceEnabler/files/9560113/xml.txt)

Beste Grüße
JoernHei

**Comment by camueller** [comment]
Genau deswegen kann die `Max. Ladeleistung` sowohl für das Fahrzeug als auch für die Wallbox angegeben werden, wobei der für das Fahrzeug angegebene Wert bevorzugt verwendet wird.

**Reply by JoernHei** [reply]
Diese habe ich den Fahrzeugen zugeordnet, nun taucht sie aber in der Anforderung nicht auf. Ich erstelle heute Abend einen Screenshot von der Einstellung im SAE.

**Reply by camueller** [reply]
Welche Anforderung?

**Reply by JoernHei** [reply]
In der Planungsanforderung über die das SEMP Protokoll an den SHM. Hier wird die maximale Leistung der Wallbox mit 22000 W angegeben. Das ist der Wert der Wallbox. Das Fahrzeug kann nur 7200 W laden, das habe ich in der Fahrzeugkonfiguration auch eigegeben.

**Reply by camueller** [reply]
Die `<MaxPowerConsumption>22000</MaxPowerConsumption>` im SEMP-Protokoll ist ein Gerätemerkmal der Wallbox und kann nicht einfach verändert werden, je nachdem, welches Fahrzeug verbunden ist. Falls der SHM dem SAE einen Leistungsempfehlung sendet, die größer ist, als die maximale Leistungsaufnahme des Fahrzeugs, dann limitiert der SAE die Leistung auf den für das Fahrzeug konfigurierten Wert.

**Reply by JoernHei** [reply]
Ok, dann hat sich meine Frage erledigt.
Ich dachte es wäre sinnvoll diese Angabe an das jeweilige Fahrzeug anzupassen aber wenn das nicht vorgesehen ist, ändere ich es bei Wechsel des Fahrzeugs in der Angabe zu der Wallbox. Vielen Dank für die schnelle Hilfe.

**Reply by th-hock** [reply]
Hallo camueller,
ich wüsste gerne wie der SHM im SEMP Request eine Leistungsempfehlung übermitteln kann. In der SEMP Spezifikation (1.0.6) die mir vorliegt kann ich in der DeviceControl Message nur das <On> Element finden, welches den Verbraucher ein- bzw. ausschaltet.
Die SMA Developer Seite ist leider auch nicht mehr erreichbar.
Vielen Dank für die Hilfe.
Th-Hock

**Reply by camueller** [reply]
Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/SEMP-41YE3814-AN-EVCharger-1.0.3.pdf

**Comment by th-hock** [comment]
Vielen herzlichen Dank für die schnelle Antwort. Genau das hatte ich gesucht :-)

Ein tolles Projekt hast du da auf die Beine gestellt. Ich bin sehr beeindruckt. Ich nutze es bislang nur nicht, weil ich meine Projekte mit einem ESP realisiere und meine Software dazu selbst schreibe.

**Reply by th-hock** [reply]
Ich habe das EV Charger Dokument mittlerweile studiert und würde gerne über das PowerLevels Element Leistungsstufen vorgeben. Ich konnte allerdings dafür keine Beispiele finden und das Dokument macht an dieser Stelle auch keine näheren Angaben. Weißt du wie ich die einzelnen Werte angeben muss? Einfach als kommaseparierte Liste? Oder gibt es eventuell noch eingeschachtelte Elemente?
Danke.

**Reply by camueller** [reply]
```
<Characteristics>
    <MaxPowerConsumption>4000</MaxPowerConsumption>
    <MinPowerConsumption>1380</MinPowerConsumption>
    <PowerLevels>
        <PowerLevel>500</PowerLevel>
        <PowerLevel>1000</PowerLevel>
        <PowerLevel>1500</PowerLevel>
    </PowerLevels>
</Characteristics>
```
